src/prefect/agent/fargate/agent.py

```diff
@@ -73,13 +84,34 @@ def __init__(
         region_name = region_name or os.getenv("REGION_NAME")
 
         # Agent task config
+        self.task_role_arn = task_role_arn or os.getenv("TASK_ROLE_ARN")
+        self.logger.debug("Task role arn {}".format(self.task_role_arn))
+
+        self.execution_role_arn = execution_role_arn or os.getenv("EXECUTION_ROLE_ARN")
+        if not self.execution_role_arn:
+            self.logger.exception(
+                "Fargate requires task definition to have execution role ARN to support ECR images"
+            )
+            raise Exception("Fargate task execution role required")
```

Comment:
I don't think this Exception needs to be raised however a warning could still be logged. The reason being (for example) in my set up I am using images based in DockerHub and this `execution_role_arn` wasn't required.

Reply:
Sounds good - I'm happy to remove the Exception and just log the warning. The reason we did this was because without the `execution_role_arn`, the iam role for the agent wasn't being set properly and this fixed it. In your aws environment is an iam role being used for the fargate tasks? Is it somehow defaulting to a iam role? (Just curious). I will update shortly.

Reply:
For mine I'm am actually not using an IAM role so using one may actually warrant the need of the `execution_role_arn` I will test various set ups / configurations and see if that will affect it. Thanks for the feedback 👍